File: Modules/SEDO/EMS/SIE.Web.EMS/SpareParts/Commands/AddStoreDetailCommand.js
SIE.defineCommand('SIE.Web.EMS.SpareParts.Commands.AddStoreDetailCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit" },
    /**
    * 是否允许执行
    * @param {any} view 当前视图
    */
    canExecute: function (view) {
        if (view.getParent()._selection != null && view.getParent()._selection.length == 1 && view.getParent()._selection[0].getStoreStatus()==0) return true; else return false;//选定行才显示添加属性按钮
    },
    onItemCreated: function (entity) {
        if (entity) {
            var model = entity.data;
            var me = this;
            this.view.execute({
                data: [],
                success: function (res) {
                    entity.setBatchNumber(res.Result);
                }
            }, me.view);
        }
    },
});

File: Modules/SEDO/EMS/SIE.Web.EMS/SpareParts/Commands/AddStoreDetailCommand.cs
using SIE.EMS.SpareParts;
using SIE.Web.Command;

namespace SIE.Web.EMS.SpareParts.Commands
{
    /// <summary>
    /// 备件入库添加命令
    /// </summary>
    [JsCommand("SIE.Web.EMS.SpareParts.Commands.AddStoreDetailCommand")]
    public class AddStoreDetailCommand : ViewCommand
    {
        /// <summary>
        /// 执行添加操作
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>工作任务号</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var No = RT.Service.Resolve<SparePartController>().GetDetailCode();
            return No;
        }
    }
}
